src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -301,9 +342,46 @@ def _process_genome_attri_tools(genome_attr_tools: set[str],
         docs.extend(parse_ops(root_dir, kbase_collection, load_ver))
 
     docs = merge_docs(docs, '_key')
+    docs = _add_upa_info(docs, root_dir, kbase_collection, load_ver, tool)
     output = f'{kbase_collection}_{load_ver}_{"_".join(genome_attr_tools)}_{COMPUTED_GENOME_ATTR_FILE_SUFFIX}'
     _create_import_files(root_dir, output, docs)
 
+    export_types_output = f'{kbase_collection}_{load_ver}_{"_".join(ENCOUNTERED_TYPES)}_{KBCALL_EXPORT_TYPES_FILE_SUFFIX}'
+    types_docs = _create_export_types_doc(kbase_collection, loader_common_names.GENOME_ATTRIBS, load_ver, ENCOUNTERED_TYPES)
+    _create_import_files(root_dir, export_types_output, [types_docs])
+
+
+def _add_upa_info(docs, root_dir, kbase_collection, load_ver, tool):
+    # Load assembly UPA data into genome attributes table, where the tool identifier is irrelevant 
+    
+    meta_lookup = {}
+    res_dict = {row[names.FLD_KBASE_ID]: row for row in docs}
+    result_dir = _locate_dir(root_dir, kbase_collection, load_ver, tool=tool)
+    batch_dirs = _get_batch_dirs(result_dir)
+    for batch_dir in batch_dirs:
+        batch_result_dir = os.path.join(result_dir, batch_dir)
+        metadata_file = os.path.join(batch_result_dir, loader_common_names.GENOME_METADATA_FILE)
+        try:
+            meta_df = pd.read_csv(metadata_file, sep='\t')
+        except Exception as e:
+            raise ValueError('Unable to retrieve the genome metadata file') from e
+        meta_dict = dict(zip(meta_df[loader_common_names.DATA_ID_COLUMN_HEADER], meta_df[loader_common_names.META_FILE_NAME]))
+        meta_lookup.update(meta_dict)
+    docs = _update_docs_with_upa_info(res_dict, meta_lookup)
```

Comment:
I would make this method just return the meta_lookup variable and then call the update docs method right after that in `_process_genome_attri_tools`. Then the update docs method can return the encountered types and pass them to the create export types method

Reply:
Updated!

Reply:
:+1: